src/app/pages/datasets/components/dataset-management/dataset-management.component.ts

```diff
@@ -20,7 +22,20 @@ import { WebSocketService } from 'app/services';
 })
 export class DatasetsManagementComponent implements OnInit {
   isLoading$ = this.datasetStore.isLoading$;
-  selectedDataset$ = this.datasetStore.selectedDataset$;
+  isLoadingDataset = false;
+  selectedDataset$ = this.datasetStore.selectedDataset$.pipe(
```

Comment:
This should be part of the state, more specifically an effect with additional properties and selectors.